src/components/header/header.js

```diff
@@ -60,13 +48,28 @@ const Header = (props) => {
 }
 
 Header.propTypes = {
-    id: PropTypes.string.isRequired
+    id: PropTypes.number
   }
 
 Header.defaultProps = {
-    id: '',
+    id: null,
+    showMovieDetails: false
 }
 
-const MemoizedHeader = React.memo(Header);
+const mapStateToProps = (state) => {
+    return {
+        movies: state.movies,
+        id: state.id,
+        showMovieDetails: state.showMovieDetails,
+    }
+  }
+  
+  const mapDispatchToProps = (dispatch) => {
```

Comment:
You can use short form `mapDispatchToProps = { hideDetails: movieHideDetails }`

Reply:
done